src/components/MessageList.jsx

```diff
@@ -0,0 +1,66 @@
+import { formatDistance } from "date-fns";
+
+export const MessageList = ({ thoughts, setThoughts }) => {
+  //console.log(thoughts,setThoughts)
+
+  // handle new likes to the API
+  const handleLikeIncrease = async (_id) => {
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+    await fetch(
+      `https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts/${_id}/like`,
+      options
+    )
+      .then((response) => response.json())
+      .then((json) => {
+        // Assuming the API response includes the updated like count
+        const updatedThoughts = thoughts.map((thought) => {
+          if (thought._id === json._id) {
+            // Update the hearts count with the new value from the API
+            thought.hearts = json.hearts;
+          }
+          return thought;
+        });
+
+        // Update the state with the new like count
+        setThoughts(updatedThoughts);
+      })
+      .catch((error) => {
+        // Handle any errors here
+        console.error("Error increasing like:", error);
+      });
```

Comment:
Interesting to simultaneously map through the thoughts for fetching corresponding increment updates on like button while submitting a new thought updating the prevThoughts list.